src/coreclr/jit/gentree.cpp

```diff
@@ -7348,6 +7348,14 @@ GenTreeVecCon* Compiler::gtNewVconNode(var_types type)
     return vecCon;
 }
 
+GenTreeVecCon* Compiler::gtNewVconNode(var_types type, void* data)
+{
+    GenTreeVecCon* vecCon = new (this, GT_CNS_VEC) GenTreeVecCon(type);
+    memset(&vecCon->gtSimdVal, 0, sizeof(simd_t));
```

Comment:
We weren't previously zeroing everything (namely because someone shouldn't be accessing uninitialized data). Do we actually need this?

Reply:
@tannergooding good question, I wonder if we should even poison it here for debug. I'm worried about the cases where we e.g. promote SIMD12 to SIMD16 so we implicitly use a value we didn't set

Reply:
`GenTreeVecCon` handles that case by ensuring promotion zeros the last element; but TYP_SIMD12 codegen in general has to be robust in the face of uninitialized data in the third element as there are many ways to upcast without guaranteeing they are cleared.

Reply:
The logic that handles emitting simd12_t constants is here, for reference: https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/instr.cpp#L863-L868